### PR TITLE
feat: support coverage ci for python sdk

### LINKED ISSUE #1971: test: add coverage report for python sdk
**Describe the feature you'd like**

The [python sdk](https://github.com/4paradigm/OpenMLDB/tree/main/python) has unit tests already, but do not have the facility to produce coverage report based on that. 

Mainly two tasks: 

-  [ ] find a coverage library that can produce coverage report and add to the python sdk project
-  [ ] use the `codecov-action` to upload coverage results in `.github/workflows/sdk.yml`

**Additional context**

### PR DESCRIPTION
**Pull request by mangoGoForward**
Signed-off-by: mango <xu.weiKyrie@foxmail.com>

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
feature

* **What is the current behavior?** (You can also link to an open issue here)
fixed #1971

* **What is the new behavior (if this is a feature change)?**
None

### CODE REVIEW

File: .github/workflows/sdk.yml [hunk 1]
```diff
@@ -129,7 +129,7 @@ jobs:
         uses: codecov/codecov-action@v2
         with:
           files: java/**/target/site/jacoco/jacoco.xml,java/**/target/scoverage.xml
-          name: coverage
+          name: maven coverage
```
Comment (aceforeverd) [inline .github/workflows/sdk.yml]: why change name ? I guess we should keep the name same between all coverage steps so codecov can merge them all finally ?

File: .github/workflows/sdk.yml [hunk 2]
```diff
@@ -285,6 +285,19 @@ jobs:
           TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
           TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
 
+      - name: generate python coverage report
+        working-directory: python
+        run: |
+          pip install coverage
+          coverage run -m unittest
```
Comment (aceforeverd) [inline .github/workflows/sdk.yml]: there is no command uniitest ? 
and the pip install directory may not added to `$PATH`, so coverage command won't found

File: .github/workflows/sdk.yml [hunk 3]
```diff
@@ -285,6 +285,19 @@ jobs:
...
+
+      - name: upload python coverage to codecov
+        uses: codecov/codecov-action@v2
+        with:
+          name: python coverage
```
Comment (aceforeverd) [inline .github/workflows/sdk.yml]: `files` options is needed, so codecov can parse and upload the coverage stats

File: .github/workflows/sdk.yml (hunk 2 above)
Comment (mangoGoForward) [reply .github/workflows/sdk.yml]: Shall we use `pytest` to generate coverage report instead of coverage?

Comment (aceforeverd) [reply .github/workflows/sdk.yml]: Yes we use pytest to run unit tests. You can run this command locally and see if there is any coverage report file generated

File: .github/workflows/sdk.yml (hunk 3 above)
Comment (mangoGoForward) [reply .github/workflows/sdk.yml]: Done

**Comment by mangoGoForward** [comment]
Hi @aceforeverd , the step of generate python coverage report failed, seems `zookeeper` can't connect.  should we add a `zookeeper` container?

**Comment by aceforeverd** [comment]
> Hi @aceforeverd , the step of generate python coverage report failed, seems `zookeeper` can't connect. should we add a `zookeeper` container?

Yes it dumps, should use `steps/ut_zookeeper.sh` to start zookeeper first, you can have a reference from the `test sqlalchemy` step, it have a script file https://github.com/4paradigm/OpenMLDB/blob/af0272cec5b9646c8b7eaa3c6d9e9dfccaf6f56d/steps/test_python.sh#L24-L25

**Comment by mangoGoForward** [comment]
should use `steps/ut_zookeeper.sh` to start zookeeper first, you can have a reference from the `test sqlalchemy` step, it have a script file
> 
> https://github.com/4paradigm/OpenMLDB/blob/af0272cec5b9646c8b7eaa3c6d9e9dfccaf6f56d/steps/test_python.sh#L24-L25

Thanks so much.

File: .github/workflows/sdk.yml [hunk 4]
```diff
@@ -285,6 +285,21 @@ jobs:
           TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
           TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
 
+      - name: generate python coverage report
```
Comment (aceforeverd) [inline .github/workflows/sdk.yml]: coverage steps should be before publish to pypi step

**Comment by aceforeverd** [comment]
> should use `steps/ut_zookeeper.sh` to start zookeeper first, you can have a reference from the `test sqlalchemy` step, it have a script file
> 
> > https://github.com/4paradigm/OpenMLDB/blob/af0272cec5b9646c8b7eaa3c6d9e9dfccaf6f56d/steps/test_python.sh#L24-L25
> 
> Thanks so much.

also test_python requires you to start the openmldb cluster, so I think you can just add a few coverage lines into `test_python.sh` for coverage

**Comment by mangoGoForward** [comment]
Hi @aceforeverd , the CI of `python-sdk` has passed

File: .github/workflows/sdk.yml [hunk 5]
```diff
@@ -275,6 +275,14 @@ jobs:
           path: |
             python/openmldb/test/pytest.xml
 
+      - name: upload python coverage to codecov
+        uses: codecov/codecov-action@v2
+        with:
+          name: python coverage
```
Comment (aceforeverd) [inline .github/workflows/sdk.yml]: name should be same with others: `coverage`

**Comment by aceforeverd** [comment]
> Hi @aceforeverd , the CI of `python-sdk` has passed

nice work. I see the coverage report but saidly it only show files in `python/test`, I guess that's because it is tested with the packaged openmdlb wheel, not from souce. we may further fix it

File: .github/workflows/sdk.yml (hunk 5 above)
Comment (mangoGoForward) [reply .github/workflows/sdk.yml]: Done

**Review by aceforeverd — APPROVED** [review]

**Review by tobegit3hub — APPROVED** [review]
LGTM

**Pull request merged**